include/ginkgo/core/preconditioner/batch_preconditioner_types.hpp

```diff
@@ -0,0 +1,72 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2021, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_PUBLIC_CORE_PRECONDITIONER_BATCH_PRECONDITIONER_TYPES_HPP_
+#define GKO_PUBLIC_CORE_PRECONDITIONER_BATCH_PRECONDITIONER_TYPES_HPP_
+
+
+#include <string>
+
+
+namespace gko {
+namespace preconditioner {
+namespace batch {
+
+
+/**
+ * Types of batch preconditioners available.
+ */
+enum Type { none, jacobi };
+
+
+const char none_str[] = "none";
+const char jacobi_str[] = "jacobi";
+
+
+/**
+ * Get a string name of an available batch preconditioner type.
+ */
+inline std::string get_string_of(Type type)
+{
+    if (type == none) {
+        return none_str;
+    } else {
+        return jacobi_str;
+    }
+}
```

Comment:
Do you need the string setup ? Can you not directly compare with the enum ?

Reply:
It's only there for possible error-reporting purposes.